src/Chapter4.hs

```diff
@@ -629,7 +677,11 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+--andM lhs rhs = (&&) <$> lhs <*> rhs
+--andM lhs rhs = lhs >>= (\y -> (rhs >>= (\x -> pure (y && x))))
+andM lhs rhs = lhs >>= (\y -> if y == False then pure False else (rhs >>= (\x -> pure (y && x))))
```

Comment:
That's the correct implementation :+1: 

Additionally, you can simplify it a bit :slightly_smiling_face: 
Notice that `True && x` always returns `x`, so no need to use the `&&` operator directly

```suggestion
andM lhs rhs = lhs >>= (\y -> if y then rhs else pure False)
```

Reply:
Ahh, this task took me some time :P It was quite a challenge truly. Thanks for the suggestion it definitely has more clarity to it.